Project2015To2017/ProjectConverter.cs

```diff
@@ -40,30 +41,27 @@ IProgress<string> progress
 			if (Path.GetExtension(target).Equals(".sln", StringComparison.OrdinalIgnoreCase))
 			{
 				progress.Report("Solution parsing started.");
-				using (var reader = new StreamReader(File.OpenRead(target)))
+				var solution = new SolutionReader().Read(target, progress);
+				if (solution == null)
 				{
-					var file = new FileInfo(target);
-					string line;
-					while ((line = reader.ReadLine()) != null)
+					yield break;
+				}
+
+				if (solution.ProjectPaths?.Count > 0)
+				{
+					foreach (var projectPath in solution.ProjectPaths)
 					{
-						if(!line.Trim().StartsWith("Project(")) {
-							continue;	
+						progress.Report("Project found: " + projectPath);
+						var fullPath = Path.Combine(solution.SolutionFolder.FullName, projectPath);
```

Comment:
Would be cool to move this path absoluting inside the SolutionReader as the fact that they are relative project paths is specific to the solution file format